src/glob.rs

```diff
@@ -234,20 +254,27 @@ pub fn read_globs_from_dir<P: AsRef<Path>>(dir: P) -> Vec<Glob> {
 }
 
 pub struct GlobMap {
-    globs: Vec<Glob>,
+    globs: HashSet<Glob>,
 }
 
 impl GlobMap {
     pub fn new() -> GlobMap {
-        GlobMap { globs: Vec::new() }
+        GlobMap { globs: HashSet::new() }
     }
 
     pub fn add_glob(&mut self, glob: Glob) {
```

Comment:
Instead of using a HashSet, maybe we can keep it slightly more simple, and check for duplicates in `add_glob()`:

```rust
// Skip duplicates
if let Some(_) = self.globs.iter().find(|g| g.glob == glob.glob && g.mime == glob.mime) {
    return;
}

self.globs.push(glob);
```

Reply:
Depends on the size of the database. I far as I can tell on my machine, it is not huge but not that small either. I would prefer a constant time insertion to a linear time insertion.

Reply:
Considering that populating the globs set is a one-off operation, I'm not overly concerned with linearity; but, sure.